Go defense website

tomtseng opened the pull request:
Update our paper website to incorporate results from defense paper.

* Header no longer displays paper title + authors, instead just says "Adversarial policies in Go". Home page now contains brief description of both papers
* Update header icons to add defense paper + defense blog using ChengCheng's icon designs
* Added both papers' citations to the footer
* Add new pages for defense results, add backlink from attack paper's page on early adversarial training results
* Navbar refactor:
  * Move navbar organization/order into new file `navbar-items.ts` instead of just listing pages in the order they appear in `content.ts` 
  * Add `NavbarItem.svelte` to handle different kinds of navbar items (ordinary link vs. dropdown menu)

tomtseng commented on sgf-viewer/public/images/go-defense-paper.svg:
Generated by using `inkscape --export-plain-svg=DST SRC_PDF`. I did this in an ubuntu devbox. When I did this on my mac it didn't save the text (i.e., it saved the text as rasters or vector curves or something) and had a bloated file size.

AdamGleave reviewed (commented):
Also started drafting edits to the content at https://docs.google.com/document/d/1dHhpCaQoHXUEXoulG7PLuAEdQkchUlZnuGjjgI3tCNQ/edit?usp=sharing

tomtseng commented on sgf-viewer/src/content.ts:
this paragraph is unchanged (just an indent change)

ccstan99 commented on sgf-viewer/src/landing-page-content.ts:
```diff
@@ -1,81 +1,59 @@
+        image: "/images/defense-blog.svg",
+        imageName: "Defense blog icon",
+        description: "Defense blog",
+        url: "https://docs.google.com/document/d/1wcLUZLXejtuCHNCSPFfDVM0peLmV85046dmYsPeYhw4/edit",
```
Final link will be: https://far.ai/post/2024-05-go-defense/ 
Current deployed preview link is: https://deploy-preview-131--alignmentfund.netlify.app/post/2024-05-go-defense/

AdamGleave commented:
Hey Tom, excited to see this PR come together! :)

Not had a chance to look at code yet, but some initial thoughts from looking at the deploy preview.

**Drop-down NavBar**: looks and works great

**Logos**
The navbar with the logos and link to paper/blog/video/etc is quite busy now. Worth thinking if there's any we can cut either the total # of elements, or just make them take up less space? Low-hanging fruit here would be to replace "Attack video" with something less wide (and more in keeping with the other logos) rather than a screenshot. E.g. here's a video icon from [NounProject](https://thenounproject.com/icon/video-2587221/) (brand colored version attached) or [FlatIcon](https://www.flaticon.com/free-icon/play_2377793?term=video&page=1&position=35&origin=search&related_id=2377793)
![noun-video-2587221-009999](https://github.com/AlignmentResearch/KataGoVisualizer/assets/1547683/c4464cec-955e-4684-a378-0cf26d44c148). We could probably also remove GitHub and just include a link to the source code in the footer but that may not be necessary.

**Game Moves**: Clicking on hyperlinks for go moves update the board but don't scroll down. I think that's the right call but it did mean they seemed like no-ops at first when there's a lot of text. Maybe we could hide the unused table rows (e.g. have it be collapsible under a "see more") for these sections to save a bit of screen space? This is definitely a nice-to-have though, only worth addressing once other more pressing things are done. 

**Todos**
Some things to not forget to address before merging:
  - [ ] Update blog post link (as needed) -- currently 404's on placeholder https://far.ai/post/2024-05-go-defense/
  - [ ] Update the citation to point to arXiv ePrint ID
  - [ ] Add strength evaluation section for vision transformer

AdamGleave commented on sgf-viewer/src/content.ts:
```suggestion
                "description": [ "The fine-tuned cyclic adversary also beats <code>b60-s7702</code> using 100,000 victim visits with a win rate of 7/40 = 17.5%. (The games displayed are non-randomly selected to show the wins achieved by the adversary.)" ]
```

AdamGleave commented on sgf-viewer/src/content.ts:
```diff
@@ -800,45 +783,429 @@ export const pages: object =
+                "max_games": 10,
+                "adversary": "Kellin Pelrine",
+                "victim": "<code>dec23-victim</code>, 1 visit",
+                "description": [
```
I made some suggestions in the GDoc you may want to port over: https://docs.google.com/document/d/1dHhpCaQoHXUEXoulG7PLuAEdQkchUlZnuGjjgI3tCNQ/edit#heading=h.suo6dcn8qh6f (sorry for editing in wrong place!)

AdamGleave commented on sgf-viewer/src/content.ts:
```diff
@@ -800,45 +783,429 @@ export const pages: object =
+                "adversary": "<code>validation-adversary</code>",
+                "victim": "<code>v</code><sub><code>9</code></sub>, 512 visits",
+                "description": [
+                    "We trained a new adversary by fine-tuning from an early adversarial checkpoint. It was able to defeat <code>v</code><sub><code>9</code></sub> at 512 visits of search in 81% of games. The win rate drops to 4% at 4096 visits, demonstrating that our victim is easily attacked until it uses high amounts of search. Explore randomly sampled games below.",
```
```suggestion
                    "We trained a new adversary by fine-tuning from an early adversarial checkpoint. It was able to defeat <code>v</code><sub><code>9</code></sub> at 512 visits of search in 81% of games, but the win rate drops to 4% at 4096 visits. This demonstrates that our victim is easily attacked until it uses high amounts of search. Explore randomly sampled games below.",
```

AdamGleave commented on sgf-viewer/src/content.ts:
```diff
@@ -800,45 +783,429 @@ export const pages: object =
+                "victim": "<code>v</code><sub><code>9</code></sub>, 512 visits",
+                "description": [
+                    "We trained a new adversary by fine-tuning from an early adversarial checkpoint. It was able to defeat <code>v</code><sub><code>9</code></sub> at 512 visits of search in 81% of games. The win rate drops to 4% at 4096 visits, demonstrating that our victim is easily attacked until it uses high amounts of search. Explore randomly sampled games below.",
+                    "This adversary starts by inducing the victim to form bamboo joints: pairs of stone separated by two empty spaces. In normal games, these are often efficient shapes due to their strong connections. In the first game here, the first bamboo joint is formed on <a class='clickable' onclick='setMove(`v9-vs-validation`, 24)'>move 24</a> between that white stone, the one next to it,  ...
```
consider adding links for "218" and "220" as well?

AdamGleave reviewed (commented):
The links to Go moves are all relative to the *current* game -- so are confusing if you end up selecting a different game from the table. Can we jump back to the original game perhaps?

I added one commit to change the Go figure width on homepage since 600px seemed too small on desktop -- please do review my change as it was a bit of a shot in the dark, but seems to look OK on both mobile & desktop with this breakpoint.

One small glitch I noticed on Chrome: if I click on a link in "Extra content", the menu remains visible, I think because it's tall enough I'm still "hovering" over the (now visible) part of the menu. This isn't triggered in the "Undefended agent" category because the menu is short enough the page scrolls away from it. Maybe we want to make on-click explicitly close the menu?

AdamGleave commented on sgf-viewer/src/content.ts:
```diff
@@ -800,45 +783,429 @@ export const pages: object =
+                "title": "Strength evaluation",
+                "dir_name": "vit-strength",
+                "description": [
+                    "TODO"
```
TODO: revise before merge

AdamGleave reviewed (changes requested):
Thanks for working on this PR Tom -- overall looks in good shape, just a few minor issues (outlined above) to be fixed plus missing content/links to update before we merge.

Note I only skimmed the new sections in `content.ts` as I'd already reviewed the GDoc it was based on.

tomtseng commented on sgf-viewer/src/components/Section.svelte:
```diff
@@ -7,7 +7,7 @@
 
 <h3 id={section["dir_name"]}>
-    {section["title"]}
+    {@html section["title"]}
```
none of the new pages but `/adversarial-training#cyclic_orig_vs_b60_s7702m_v1` benefits from it

tomtseng commented on streamlit_app/tests/data/test_games.sgfs:
```diff
@@ -1,8 +1,8 @@
-(;FF[4]GM[1]SZ[19]PB[t0-s108263680-d27265989__victim]PW[victim]HA[0]KM[6.5]RU[koPOSITIONALscoreAREAtaxNONEsui1]RE[B+12.5]C[startTurnIdx=0,initTurnNum=0,gameHash=B5B369DE905FCC115D2378B92660E883,gtype=normal];B[bb]C[0.39 0.61 0.00 -9.3 v=600 weight=0.75];W[qd]C[0.75 0.25 0.00 6.1 v=32 weight=0.66];B[rr]C[0.36 0.64 0.00 -10.0 v=600 weight=0.80];W[cp]C[0.93 0.07 0.00 13.4 v=32 weight=0.85];B[rb]C[0. ...
```
ok sounds good, I ran the sgf-viewer with these game_info.py changes on the original cyclic training run and nothing crashed so we should be ok unless we start digging up some old runs for some reason

tomtseng commented:
Addressed most comments, still not done with:
* TODOs for defense paper + defense paper links since they aren't up yet
* when clicking move link, should also switch to the correct game (this needs a refactor because there are several interacting pieces here, i'll make it a separate PR and probably won't get it done by neurips deadline)

AdamGleave commented on sgf-viewer/src/content.ts:
```diff
@@ -1079,16 +1098,50 @@ export const pages: object =
+                "victim": "<code>ViT-victim</code>, 65536 visits, 64 search threads",
+                "description": [
+                    "After a close early game between ViT (black) and Yang (white), a complicated fight develops on the left side and center, starting around <a class='clickable' onclick='setMove(`vit-vs-yang`, 52)'>move 52</a>. According to KataGo, this move would have been better as a cap two lines to the right of black's stone on the middle of the left side, or one line below the cap. If so, t ...
+                    "Nonetheless, after these moves the game remained close. The most deciding move was that White should have played <a class='clickable' onclick='setMove(`vit-vs-yang`, 66)'>66</a> directly below white's leftmost stone, to probe how black connects the two stones above. If black plays the best, empty triangle connection, then a complicated ko situation may develop but the game re ...
```
We don't capitalize white elsewhere so suggest not capitalizing it here for consistency (or change it everywhere)
```suggestion
                    "Nonetheless, after these moves the game remained close. The most deciding move was that white should have played <a class='clickable' onclick='setMove(`vit-vs-yang`, 66)'>66</a> directly below white's leftmost stone, to probe how black connects the two stones above. If black plays the best, empty triangle connection, then a complicated ko situation may develop but the game remains only marginally favored for black, whereas other connections will tilt the game in white's favor. But after <a class='clickable' onclick='setMove(`vit-vs-yang`, 66)'>66</a> in the game, black threatens to cut off a stone with <a class='clickable' onclick='setMove(`vit-vs-yang`, 67)'>67</a>, and the timing to probe the connection on the left is lost. In the end, with <a class='clickable' onclick='setMove(`vit-vs-yang`, 92)'>92</a> white is able to live, but black's position on the outside is better and black has a decent advantage.",
```

AdamGleave commented on sgf-viewer/src/content.ts:
```diff
@@ -1079,16 +1098,50 @@ export const pages: object =
-                    "TODO"
+                    "After ViT's big loss in the flying dagger joseki in the preceding game, Li agreed to play a game where he would avoid that pattern. In this game, on <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 11)'>move 11</a>, black (ViT) offers a chance to initiate the flying dagger pattern, but white (Li) declines by playing at <a class='clickable' onclick='setMove(`vit-vs-l ...
+                    "The game then remains balanced for many moves, with both sides playing well and fighting back and forth over narrow margins. On <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 71)'>move 71</a>, black should have continued in the top right but makes a significant mistake by playing on the bottom side. This allows white to guarantee the safety of the top right throug ...
+                    "Over the following moves, black gradually reduces white's advantage, and then on <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 118)'>move 118</a> white misses some opportunities on the lower side and center, making the game virtually even again. <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 131)'>Move 131</a> is a nice tesuji and the only way for bla ...
```
```suggestion
                    "Over the following moves, black gradually reduces white's advantage, and then on <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 118)'>move 118</a> white misses some opportunities on the lower side and center, making the game virtually even again. <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 131)'>Move 131</a> is a nice tesuji and the only way for black to save the corner cleanly. White begins to develop a small lead again step-by-step, but loses it with <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 152)'>152</a>, which misses the timing to reduce the center after black reinforces with <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 153)'>153</a>. Finally, <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 164)'>164</a> misses a chance to play a complicated sequence of reducing moves on the bottom and center. With <a class='clickable' onclick='setMove(`vit-vs-li-rematch`, 165)'>165</a>, black fixes the largest remaining gap in the center. Thus, after a long game where white was sometimes even and sometimes favored, black (ViT) was able to close the game in the end."
```

AdamGleave commented on sgf-viewer/src/content.ts:
```diff
@@ -1187,11 +1240,23 @@ export const pages: object =
+                "adversary": "Kellin Pelrine (author)",
+                "victim": "<code>ViT-victim</code>, 65536 visits, 64 search threads",
+                "description": [
+                    "A Go expert (Kellin Pelrine) was also able to use a cyclic attack to beat our ViT-victim. The game is shown below. Through <a class='clickable' onclick='setMove(`vit-vs-human-cyclic`, 56)'>move 56</a>, White (Kellin) sets up a loosely surrounded square group destined to be the inside of the cyclic group. This follows the shape used in some of the wins of our original cyclic a ...
```
```suggestion
                    "A Go expert (Kellin Pelrine) was also able to use a cyclic attack to beat our ViT-victim. The game is shown below. Through <a class='clickable' onclick='setMove(`vit-vs-human-cyclic`, 56)'>move 56</a>, white (Kellin) sets up a loosely surrounded square group destined to be the inside of the cyclic group. This follows the shape used in some of the wins of our original cyclic adversary against this victim. In subsequent moves, white gradually surrounds the cyclic group, with a particular focus on making sure the surrounding groups have lots of liberties so that black will have to see its group is in danger early on to save it. Beginning around <a class='clickable' onclick='setMove(`vit-vs-human-cyclic`, 176)'>move 176</a>, white fills in the cyclic group's liberties, as well as pressing black to make the last connections that complete the cyclic shape. Finally, after <a class='clickable' onclick='setMove(`vit-vs-human-cyclic`, 208)'>208</a>, black's fate is sealed. Black (ViT) plays on several more moves hoping for white to make a mistake, but ultimately resigns."
```

AdamGleave reviewed (approved):
LGTM, a few minor suggested changes.

Only thing we _need_ to address before merge is updating the links to point to a copy of the paper, and fixing the citation.

Addressing jumping to correct game I think could be deferred to another PR, I'll open an issue to track it.
